eslint.config.js

```diff
@@ -1,44 +1,57 @@
-import { fixupConfigRules } from '@eslint/compat';
-import pluginJs from '@eslint/js';
-import * as reactQuery from '@tanstack/eslint-plugin-query';
-import pluginImport from 'eslint-plugin-import';
-import pluginReactConfig from 'eslint-plugin-react/configs/recommended.js';
-import * as reactHooks from 'eslint-plugin-react-hooks';
 import globals from 'globals';
-import tseslint from 'typescript-eslint';
+import eslint from '@eslint/js';
+import ts from 'typescript-eslint';
+import reactRefresh from 'eslint-plugin-react-refresh';
+import reactHooks from 'eslint-plugin-react-hooks';
+import importPlugin from 'eslint-plugin-import';
+import reactPlugin from 'eslint-plugin-react';
+
+import * as tanstackQuery from '@tanstack/eslint-plugin-query';
 
 export default [
-  { languageOptions: { globals: globals.browser } },
-  pluginJs.configs.recommended,
-  ...tseslint.configs.recommended,
-  ...fixupConfigRules(pluginReactConfig),
+  eslint.configs.recommended,
+  ...ts.configs.recommended,
+  {
+    ignores: ['dist/**/*', 'node_modules/**/*', 'eslint.config.js'],
+  },
   {
+    files: ['*.ts', '*.tsx'],
+    languageOptions: {
+      ecmaVersion: 2023,
+      sourceType: 'module',
+      globals: {
+        ...globals.browser,
+      },
+      parserOptions: {
+        ecmaFeatures: {
+          jsx: true,
+        },
+      },
+    },
     plugins: {
-      '@tanstack/query': reactQuery,
-      import: { rules: pluginImport.rules },
-      'react-hooks/recommended': reactHooks,
+      react: reactPlugin,
+      'react-hooks': reactHooks,
+      '@tanstack/query': {
+        rules: tanstackQuery.rules,
+      },
+      import: { rules: importPlugin.rules },
     },
     rules: {
-      '@tanstack/query/exhaustive-deps': 'error',
-      '@tanstack/query/no-rest-destructuring': 'warn',
-      '@tanstack/query/stable-query-client': 'error',
-      'import/exports-last': 'off',
-      'import/no-useless-path-segments': [
-        'error',
-        {
-          noUselessIndex: true,
-        },
-      ],
-      'import/order': [
-        'error',
-        {
-          alphabetize: { caseInsensitive: true, order: 'asc' },
-          groups: ['index', 'sibling', 'parent', 'internal', 'external', 'builtin', 'object', 'type'],
-          'newlines-between': 'always-and-inside-groups',
-        },
-      ],
```

Comment:
This code should be included

Reply:
>discussion about rules we decide what to include and what not.

Exactly. That is why i did not add so specific rules. Becuase during such discussion we will decide what to INCLUDE or NOT. Espiecially when you admitted that one of this rules... you don't know/remember why `'import/exports-last'` has been added.

Reply:
Let's do this, that way:
1. merge this - as it makes linter working
2. make workflow with automatic linter working
3. adjust linter rules

Reply:
Ok. But the linter was working before that. The rule to not warn when using `any` was turned off. And react-refresh errors were not installed because we already have this in vite. Don't know where it come from that linter have not worked before

Reply:
> Espiecially when you admitted that one of this rules... you don't know/remember why `'import/exports-last'` has been added.

I said that I don't remember in which case I've added this rule. I do know why I have done this

Reply:
> The rule to not warn when using any was turned off
For me it is totally unacceptable way to setup the linter for typescript, `any` should be prohibited and is allowed only in very very special occasions - like integrating with some lib with very bad typing or something like that, that would force us to use it and we don't have any impact to change this cause.

> Espiecially when you admitted that one of this rules... you don't know/remember why 'import/exports-last' has been added.

I agree with @swierzbicki that it's better to use it

And beside that - first we need to have checks working on gh actions, then we can focus on making good standard linter options (by standard I mean those related to react or typescript) and some more specific options. But this should be agreed in a whole team with respect to already used configuration, and should be applied to all the repos we have, to make the transition once and forgot about it.
That's why I want those 3 points I mentioned earlier to happen in presented order.